src/node_http2.h

```diff
@@ -1121,6 +1164,10 @@ class Http2Session::Http2Ping : public AsyncWrap {
 
   size_t self_size() const override { return sizeof(*this); }
 
+  uint64_t cost() const {
+    return sizeof(Http2Session::Http2Ping);
+  }
```

Comment:
I think these `cost()` methods are essentially exactly the same as `self_size()`, so it might be nicer to keep using the existing methods…?

Reply:
was considering that, kept them separate for now in case there were other bits that needed to be accounted for at some point. For instance, the cost() of a settings frame is the size of the object + the number of settings included.

Reply:
> For instance, the cost() of a settings frame is the size of the object + the number of settings included.

I think that’s a bug in the `self_size()` methods for settings then – we’re generally not very strict about `self_size()` being accurate, but as long as the semantics are “the size of the memory owned by this object”, they seem pretty much identical for both methods?

Reply:
good point... hmm ok